File: src/main/java/tech/bobliu/assignment08/repository/ThreadRepository.java
package tech.bobliu.assignment08.repository;

import java.util.List;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import tech.bobliu.assignment08.model.Thread;

public interface ThreadRepository extends JpaRepository<Thread, Long> {
    @Override
    @EntityGraph(attributePaths = {"sender"})
    List<Thread> findAll();
}

File: src/main/kotlin/tech/bobliu/assignment08/repository/ThreadRepository.kt
package tech.bobliu.assignment08.repository

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import tech.bobliu.assignment08.model.Thread

interface ThreadRepository : JpaRepository<Thread, Long> {
    @EntityGraph(attributePaths = ["sender"])
    override fun findAll(): List<Thread>
}
